### PR TITLE
 [FSTORE-1248] Papermill integration

### PR DESCRIPTION
**Pull request by marcopellegrinoit**
- Added Jupyter Notebook guide (Papermill)
- Updated Python, PySpark, and Spark job guides to 3.8.0

### CODE REVIEW

File: docs/user_guides/projects/jobs/notebook_job.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+# How To Run A Jupyter Notebook Job
+
+## Introduction
+
+All members of a project in Hopsworks can launch the following types of applications through a project's Jobs service:
+
+- Python (*Hopsworks Enterprise only*)
+- Apache Spark
+
+Launching a job of any type is very similar process, what mostly differs between job types is
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/notebook_job.md:14]: Launching a job is a similar process for Python/Spark/Flink, with the differences being
the various configuration parameters for each framework/language. After following this guide you will be able to create a job using a Jupyter Notebook.

File: docs/user_guides/projects/jobs/notebook_job.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+---
+description: Documentation on how to configure and execute a Jupyter Notebook job on Hopsworks.
+---
+
+# How To Run A Jupyter Notebook Job
```
[the remark this answers is not in the document]
Comment (jimdowling) [reply docs/user_guides/projects/jobs/notebook_job.md:5]: I see you are using the term 'Jupyter Notebook Job' consistently, so disregard my last comment.

File: docs/user_guides/projects/jobs/notebook_job.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+The image below shows the Jobs overview page in Hopsworks and is accessed by clicking `Jobs` in the sidebar.
+
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/jobs/jobs_overview.png" alt="Jobs overview">
+    <figcaption>Jobs overview</figcaption>
+  </figure>
+</p>
+
+### Step 2: Create new job dialog
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/notebook_job.md:35]: Create new job dialog ->
New job dialog

File: docs/user_guides/projects/jobs/notebook_job.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/jobs/jobs_select_python.gif" alt="Select Python job type">
+    <figcaption>Select Python job type</figcaption>
+  </figure>
+</p>
+
+### Step 4: Set the script
+
+Next step is to select the Jupyter Notebook to run. You can either select `From project`, if the file was previously uploaded to Hopsworks, or `Upload new file` which lets you select a file from your local filesystem as demonstrated below. By default, the job name is the same as the file name, but you can customize it as shown.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/notebook_job.md:59]: Next step is to select ->
Select the ...

(Given you have step numbers, you don't need to write "The next step is to ..."

File: docs/user_guides/projects/jobs/notebook_job.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/jobs/configure_py.png" alt="Set the job type">
+    <figcaption>Set the job type</figcaption>
+  </figure>
+</p>
+
+### Step 7: Execute the job
+
+Now click the `Run` button to start the execution of the job. You will be redirected to the `Executions` page where you can see the list of all executions.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/notebook_job.md:101]: where you can see the list of all executions for the job.

File: docs/user_guides/projects/jobs/notebook_job.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+notebook_job_config['appPath'] = uploaded_file_path
+
+job = jobs_api.create_job("notebook_job", notebook_job_config)
+
+```
+
+### Step 3: Execute the job
+
+In this code snippet, we execute the job with arguments and wait until it reaches a terminal state.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/notebook_job.md:161]: wait until it terminates.

File: docs/user_guides/projects/jobs/pyspark_job.md [hunk 7]
```diff
@@ -45,29 +45,36 @@ To configure a  job, click `Advanced options`, which will open up the advanced c
   </figure>
 </p>
 
-### Step 3: Set the script
+### Step 3: Set the job type
 
-Next step is to select the program to run. You can either select `From project`, if the file was previously uploaded to Hopsworks, or `Upload new file` which lets you select a file from your local filesystem as demonstrated below.
+By default, the dialog will create a Spark job. Make sure `SPARK` is chocen.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/pyspark_job.md:50]: chosen

File: docs/user_guides/projects/jobs/python_job.md [hunk 8]
```diff
@@ -43,9 +43,20 @@ By default, the dialog will create a Spark job. To instead configure a Python jo
...
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/jobs/jobs_select_python.gif" alt="Select Python job type">
+    <figcaption>Select Python job type</figcaption>
+  </figure>
+</p>
+
+### Step 4: Set the script
+
+Next step is to select the python script to run. You can either select `From project`, if the file was previously uploaded to Hopsworks, or `Upload new file` which lets you select a file from your local filesystem as demonstrated below. By default, the job name is the same as the file name, but you can customize it as shown.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/python_job.md:59]: Next step is to select the python script to run. ->
Select the python script to run.

File: docs/user_guides/projects/jobs/python_job.md [hunk 9]
```diff
@@ -80,13 +92,12 @@ It is possible to also set following configuration settings for a `PYTHON` job.
   </figure>
 </p>
 
-### Step 6: Execute the job
+### Step 7: Execute the job
 
-Now click the `Run` button to start the execution of the job, and then click on `Executions` to see the list of all executions.
+Now click the `Run` button to start the execution of the job. You will be redirected to the `Executions` page where you can see the list of all executions.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/python_job.md:97]: list of all executions for the job

File: docs/user_guides/projects/jobs/spark_job.md [hunk 10]
```diff
@@ -38,9 +38,13 @@ To configure a  job, click `Advanced options`, which will open up the advanced c
   </figure>
 </p>
 
-### Step 3: Set the jar
+### Step 3: Set the job type
 
-Next step is to select the program to run. You can either select `From project`, if the file was previously uploaded to Hopsworks, or `Upload new file` which lets you select a file from your local filesystem as demonstrated below. After that set the name for the job.
+By default, the dialog will create a Spark job. Make sure `SPARK` is chocen.
+
+### Step 4: Set the jar
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/spark_job.md:45]: Set the jar file

File: docs/user_guides/projects/jobs/spark_job.md [hunk 11]
```diff
@@ -38,9 +38,13 @@ To configure a  job, click `Advanced options`, which will open up the advanced c
...
+
+Next step is to select the program to run. You can either select `From project`, if the file was previously uploaded to Hopsworks, or `Upload new file` which lets you select a file from your local filesystem as demonstrated below. After that set the name for the job. By default, the job name is the same as the file name, but you can customize it here.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/spark_job.md:47]: Next step is to select the program to run. ->
Select the program to run.

File: docs/user_guides/projects/jobs/spark_job.md [hunk 12]
```diff
@@ -49,30 +53,32 @@ Next step is to select the program to run. You can either select `From project`,
   </figure>
 </p>
 
-### Step 4: Set the job type
+### Step 5: Set the main class
 
-Next step is to set the job type to `SPARK` to indicate it should be executed as a spark job. Then specify [additional configuration](#step-5-optional-additional-configuration) or click `Create New Job` to create the job.
+Next step is to set the main class for the application. Then specify [advanced configuration](#step-6-optional-advanced-configuration) or click `Create New Job` to create the job.
```
Comment (jimdowling) [inline docs/user_guides/projects/jobs/spark_job.md:58]: Next step is to set  -> Se

**Review by jimdowling — APPROVED** [review]
Mostly typos. Please fix before merging.

File: docs/assets/images/guides/jobs/create_new_job.png
[the remark this answers is not in the document]
Comment (marcopellegrinoit) [reply docs/assets/images/guides/jobs/create_new_job.png:1]: Yes, Papermill is only for Python jupyter notebooks.
This screenshot is just to show the initial popup when creating a job. The next step, as it was before in the guides, tells the user to select the Python radio button.
I can merge it into 1 step if you find it more intuitive.

**Pull request merged**